sdk/template/template/test/internal/beforeAfter.spec.ts

```diff
@@ -0,0 +1,120 @@
+  describe.skip("Discouraged example of `before` and `after`", function() {
+    let state: {
+      properties?: any;
+    } = {};
+    let client: InternalClass;
+
+    before(() => {
+      state = {
+        properties: {}
+      };
+
+      // The recorder needs to be initialized before the clients are created,
+      // so assume the recorder is being initialized here.
+
+      client = new InternalClass();
+      // And other global setups...
+    });
+
+    after(() => {
+      delete state.properties;
+      // And other global cleanups...
+    });
+
+    it("A test for the discouraged example of `before`", function() {
+      assert.exists(state.properties);
+      assert.exists(client);
+    });
+  });
+
+  // Use `beforeEach` to execute tasks that will prepare the resources needed
+  // for each test to run cleanly, and `afterEach` to tear down or clean
+  // those settings before the next test runs.
+  describe("Encouraged example of `beforeEach` and `afterEach`", function() {
+    let client: InternalClass;
+    let state: {
+      fruits?: string[];
+    } = {};
+
+    beforeEach(function() {
+      client = new InternalClass();
+      state = {
+        fruits: []
+      };
+
+      // And other per-test setups...
+    });
+
+    afterEach(function() {
+      // Fruits are overwritten in the beforeEach,
+      // but otherwise could be cleared up here:
+      state.fruits = [];
+
+      // And other per-test cleanups...
+    });
+
+    it("A test for the encouraged example of `beforeEach` and `afterEach`", function() {
+      assert.exists(client);
+    });
+  });
+
+  // Use `before` to declare heavy resources that can be used by more than one test.
+  // Like a stateless web server...
+  // (This test won't run if we're in a browser,
+  // since HTTP servers can't be created in the browsers).
+  if (isNode) {
+    describe("Encouraged example of `before` and `after`", function() {
+      const expectedHttpResponse = "Hello World!";
+      let server: any;
+
+      /**
+       * helloWorldRequest makes a get request to the env.SERVER_ADDRESS
+       * and returns a promise that resolves when the server responds.
+       */
+      async function helloWorldRequest(): Promise<string> {
+        return new Promise((resolve) => {
+          const http = require("http");
+          http.get("http://localhost:8080", (res: any) => {
+            let data = "";
+            res.on("data", (chunk: string) => {
+              data += chunk;
+            });
+            res.on("end", () => {
+              resolve(data);
+            });
+          });
+        });
+      }
+
+      before(function() {
+        // Only internal tests may use mocks of servers.
+        const http = require("http");
+        server = http.createServer(function(_: any, res: any) {
+          res.write(expectedHttpResponse);
+          res.end();
+        });
+        server.listen(8080);
```

Comment:
I understand what you mean. My main interest what to show how to create a common stateless resource, and then how to close it. A web server is the first thing that came to mind. It's also worth pointing out that this is an internal test and mocks are not discouraged in internal tests.

Reply:
What other "something" can I use to represent creating a reusable stateless resource and then disposing of it? I'm open to ideas!

Reply:
This isn't a mock though, is it? I thought this is using the real http node built in, which will bind a port on the machine, which prevents a test like this from being parallelized.

Maybe another example of this would be a fake File instance that is backed by a Buffer?

Reply:
Sounds good! I'll do that 🖖 Thank you!

Reply:
Done!